include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -315,3 +315,53 @@ def MuxOp : RTLOp<"mux",
     }]>
   ];
 }
+
+def BitArrayType : Type<CPred<[{
+  $_self.isa<ArrayType>() &&
+  $_self.cast<ArrayType>().getElementType().isSignlessInteger(1)
```

Comment:
This would be more useful if it allowed casting to arrays of any-width integers.  you could do:
i64 -> 8xi8

Reply:
That would allow lists of numbers to be cast'd as well. The problem is that we don't differentiate between bits/bytes/nibbles/words and integers. Perhaps we should create a `word` type that only allows power of 2 widths and you can't do anything with except cast?

Reply:
As long as the sizeof(array) == sizeof(othertype), that's fine.
i64 -> 11xi6 would be rejected.

Reply:
I don't think we are looking at arbitrary casts between dialects (as discussed in your link), we are looking at the equivalent of reinterpret_cast between RTL synthesizable types, which all have known length.  This is pretty close to a subset of the SV streaming operator.  The cast form packed struct/array <-> integerType is going to be needed to handle verilog parsing, as will the streaming operator, so I'm just proposing being general enough to handle those.

Reply:
I hear ya. If we go with this as-is, you could just use two ops. Given the fact that the RTL dialect is untyped, I'm willing to allow conversions between RTL/SV types of equal bitwidths.

I was trying to be as restrictive as possible given @lattner's (justified) reaction to my first (stewpot) PR. I would prefer that he weigh in before I make any changes.